### PR TITLE
Include gl3d scene.aspectmode changes in relayout updates

### PR DESCRIPTION
**Pull request by archmoj**
Follow up of #4578 this PR adds gl3d `scene.aspectmode` changes in relayout updates.

[Demo](https://codepen.io/MojtabaSamimi/pen/ExjPNRV?editors=0010)

@plotly/plotly_js

### CODE REVIEW

File: src/plots/gl3d/scene.js [hunk 1]
```diff
@@ -219,6 +219,12 @@ proto.initializeGLPlot = function() {
         if(scene.isAspectChanged(layout)) {
             // scene updates
             update[scene.id + '.aspectratio'] = scene.glplot.getAspectratio();
+
+            if(layout[scene.id].aspectmode !== 'manual') {
```
Comment (etpinard) [inline src/plots/gl3d/scene.js]: How do we get back to the original `aspectmode` ?

Comment (archmoj) [reply src/plots/gl3d/scene.js]: Should we?

Comment (etpinard) [reply src/plots/gl3d/scene.js]: When resetting the view (e.g via the modebar buttons), yes.

Comment (archmoj) [reply src/plots/gl3d/scene.js]: Good call. I think we have some tests related to that.

Comment (archmoj) [reply src/plots/gl3d/scene.js]: https://github.com/plotly/plotly.js/blob/4d24c36c93d1d1cca4ad4a195cbaf43f53e963fe/test/jasmine/tests/plot_api_test.js#L394-L411

Comment (etpinard) [reply src/plots/gl3d/scene.js]: So, to be clear, once a user scrolls into a orthographic scene, the `aspectmode` is turned to `'manual'` and the only way to get back to original `aspectmode` is by resetting the view?

Comment (archmoj) [reply src/plots/gl3d/scene.js]: Or we should keep that in `viewInitial` props?

Comment (etpinard) [reply src/plots/gl3d/scene.js]: How is it handled currently?

Comment (archmoj) [reply src/plots/gl3d/scene.js]: `apsectmode` added to `viewInitial` in https://github.com/plotly/plotly.js/pull/4579/commits/1fdc49c7c22dc3b73f7a0f3572b675ea2457b5b7 and tests are added in https://github.com/plotly/plotly.js/pull/4579/commits/6ae280967e5e2e91762d28cc77290516735efc83.

File: src/plots/gl3d/scene.js [hunk 2]
```diff
@@ -790,6 +790,9 @@ proto.plot = function(sceneData, fullLayout, layout) {
             y: fullSceneLayout.aspectratio.y,
             z: fullSceneLayout.aspectratio.z
         };
+
+        // also keep track of aspectmode here
+        scene.viewInitial.aspectmode = fullSceneLayout.aspectmode;
```
Comment (etpinard) [inline src/plots/gl3d/scene.js]: Shouldn't this be

```js
if(!scene.viewInitial.aspectmode) {
  scene.viewInitial.aspectmode = fullSceneLayout.aspectmode
}
```

that way we don't override `viewInitial` on updates?

Comment (archmoj) [reply src/plots/gl3d/scene.js]: Good call.
We could. Done in https://github.com/plotly/plotly.js/pull/4579/commits/c730f1276ad167adf44b13cf8482fb317d88843c.

**Comment by etpinard** [comment]
Thanks - :dancer: !

**Pull request merged**